File: adaApi/src/main/java/br/com/adaApi/api/service/UserService.java
package br.com.adaApi.api.service;

import java.util.List;

import org.springframework.data.domain.Page;

import br.com.adaApi.api.entity.User;
import br.com.adaApi.api.enums.ProfileEnum;

public interface UserService {
	
	User findByEmail(String email);
	
	User createOrUpdate(User user);
	
	User findById(Long id);
	
	void delete(Long id);
	
	Page<User> findAll(int page, int count);

	List<User> findUserParams(String email, ProfileEnum profile);

}

File: adaFront/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { User } from '../model/user.model';
import { URL_API } from './url.api';
import { SharedService } from './shared.service';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) {}

  login(user: User) {
    return this.http.post(`${URL_API}/api/auth`, user);
  }

  createOrUpdate(user: User) {
    if ( user.id !== null && user.id !== '') {
      return this.http.put(`${URL_API}/api/user`, user);
    } else {
      user.id = null;
      return this.http.post(`${URL_API}/api/user`, user);
    }
  }

  findAll(page: number, count: number) {
    return this.http.get(`${URL_API}/api/user/${page}/${count}`);
  }

  findById(id: string) {
    return this.http.get(`${URL_API}/api/user/${id}`);
  }

  delete(id: string) {
    return this.http.delete(`${URL_API}/api/user/${id}`);
  }

  getUser() {
    return this.http.get(`${URL_API}/api/getuser/${SharedService.getInstance().token}`);
  }

}
